src/act-compat.js

```diff
@@ -0,0 +1,12 @@
+import {act as reactAct} from 'react-dom/test-utils'
+
+// act is supported react-dom@16.8.0
+// and is only needed for versions higher than that
+// so we do nothing for versions that don't support act.
+const act = reactAct || (cb => cb())
+
+function rtlAct(...args) {
+  return act(...args)
```

Comment:
you can actually polyfill it for older versions of react 
```jsx
function act(fn){
  ReactDOM.unstable_batchedUpdates(fn);
  ReactDOM.render(<div/>, document.createElement('div'));
}
```

you won't get the warnings, but something's better than nothing I suppose.